examples/simple/src/main/java/server/userauth/SimpleTokenFactory.java

```diff
@@ -90,9 +90,8 @@ public MslError acceptNonReplayableId(final MslContext ctx, final MasterToken ma
         
         // Accept if there is no non-replayable ID.
         final String key = masterToken.getIdentity() + ":" + masterToken.getSerialNumber();
-        final Long largestNonReplayableId = nonReplayableIds.get(key);
+        final Long largestNonReplayableId = nonReplayableIds.putIfAbsent(key, nonReplayableId);
```

Comment:
If there is a non-null value associated with that key then wouldn't putIfAbsent return null?

Reply:
No, putIfAbsent returns "the previous value associated with the specified key, or null if there was no mapping for the key."

Reply:
OK, I re-read the documentation and agree that you are correct.